File: Level1/체육복.py
n = 7
lost = [1, 4, 5, 7]
reserve = [2, 3]


def solution(n, lost, reserve):
    list = [1] * n
    for i in reserve:
        list[i-1] = 2
    for i in lost:
        if i not in reserve:
            list[i-1] = 0
        else:
            list[i-1] = 1
    print(list)
    for i in range(len(list)):
        if i == 0:
            if list[i] == 0 and list[i+1] == 2:
                list[i] = 1
                list[i+1] = 1
        if i == (len(list) - 1):
            if list[i] == 0 and list[i-1] == 2:
                list[i] = 1
                list[i-1] = 1
        else:
            if list[i] == 0 and list[i-1] == 2:
                list[i] = 1
                list[i-1] = 1
            elif list[i] == 0 and list[i+1] == 2:
                list[i] = 1
                list[i+1] = 1
    print(list)
    return n - list.count(0)




did = solution(n, lost, reserve)
print(did)

File: Level1/체육복.java
import java.util.*;
class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        Arrays.sort(lost);
        Arrays.sort(reserve);
        
        boolean [] able = new boolean[reserve.length];
        Arrays.fill(able, true);
        boolean[] can = new boolean[lost.length];
        
        int answer = n - lost.length;
        for(int i=0; i<lost.length; i++) {
            for(int j=0; j<reserve.length; j++) {
                if(lost[i] == reserve[j]) {
                    answer++;
                    can[i] = true;
                    able[j] = false;
                    break;
                }
            }
        }

        for(int i=0; i<lost.length; i++) {
            if(!can[i]) {
                int loster = lost[i];
                for(int j=0; j<reserve.length; j++) {
                    if(able[j]) {
                        if(loster + 1 == reserve[j] || reserve[j] == loster-1) {
                            able[j] = false;
                            answer++;
                            break;
                        }
                    }
                }
            }
        }
        
        return answer;
    }
}
